server/util.js

```diff
@@ -25,7 +35,40 @@ function httpRequest(url) {
     });
 }
 
+function getUserWithDefaultsSet(user) {
+    let userToReturn = user;
+
+    if(!userToReturn) {
+        return userToReturn;
+    }
+
+    if(!userToReturn.settings) {
+        userToReturn.settings = {
+            disableGravatar: false,
```

Comment:
This approach isn't granular enough. For example, if you add a new default under `settings` but the user already has the settings object, the new value won't be there. Instead of checking that they're missing a particular value, you could use `_.extend` to setup the defaults and copy over them with the user's existing values. E.g.
```javascript
// The if goes away, since _.extend(anything, undefined) === anything
userToReturn.settings = _.extend({}, { disableGravatar: false, windowTimer: 10 }, userToReturn.settings);
```

Unfortunately, `extend` isn't recursive so you can't provide all the defaults at once due to the way our settings are nested. Could probably write a recursive version without huge trouble, but not sure it's worth it for only having 4 keys.